nilearn/plotting/displays.py

```diff
@@ -350,6 +350,115 @@ def _add_lines(self, line_coords, line_values, cmap,
 
+class GlassBrainHemisphericAxes(GlassBrainAxes):
+    """An MPL axis-like object that displays a 2D projection of 3D
+    volumes with a schematic view of the brain. In the case of saggital cuts,
+    projects only data in the selected left or right hemisphere.
+
+    """
+    def __init__(self, ax, direction, coord, plot_abs=True, **kwargs):
+        super(GlassBrainHemisphericAxes, self).__init__(
+            ax, direction, coord, plot_abs, **kwargs)
+
+    def draw_2d(self, data_2d, data_bounds, bounding_box,
+                type='imshow', **kwargs):
+        # kwargs massaging
+        kwargs['origin'] = 'upper'
+
+        if self.direction == 'y':
+            (xmin, xmax), (_, _), (zmin, zmax) = data_bounds
+            (xmin_, xmax_), (_, _), (zmin_, zmax_) = bounding_box
+        elif self.direction in 'lr':
+            (_, _), (xmin, xmax), (zmin, zmax) = data_bounds
+            (_, _), (xmin_, xmax_), (zmin_, zmax_) = bounding_box
+        elif self.direction == 'z':
+            (xmin, xmax), (zmin, zmax), (_, _) = data_bounds
+            (xmin_, xmax_), (zmin_, zmax_), (_, _) = bounding_box
+        else:
+            raise ValueError('Invalid value for direction %s' %
+                             self.direction)
+        ax = self.ax
+        # Here we need to do a copy to avoid having the image changing as
+        # we change the data
+        im = getattr(ax, type)(data_2d.copy(),
+                               extent=(xmin, xmax, zmin, zmax),
+                               **kwargs)
+
+        self.add_object_bounds((xmin_, xmax_, zmin_, zmax_))
+
+        return im
+
+    def draw_left_right(self, size, bg_color, **kwargs):
+        if self.direction in 'lr':
+            return
+        ax = self.ax
+        ax.text(.1, .95, 'L',
+                transform=ax.transAxes,
+                horizontalalignment='left',
+                verticalalignment='top',
+                size=size,
+                bbox=dict(boxstyle="square,pad=0",
+                          ec=bg_color, fc=bg_color, alpha=1),
+                **kwargs)
+
+        ax.text(.9, .95, 'R',
+                transform=ax.transAxes,
+                horizontalalignment='right',
+                verticalalignment='top',
+                size=size,
+                bbox=dict(boxstyle="square,pad=0", ec=bg_color, fc=bg_color),
+                **kwargs)
+
+    def transform_to_2d(self, data, affine):
+        """ Returns the maximum of the absolute value of the 3D volume
+            along an axis.
+
+            Parameters
+            ==========
+            data: 3D ndarray
+                The 3D volume
+            affine: 4x4 ndarray
+                The affine of the volume
+
+        """
+        if self.direction in 'lr':
+            max_axis = 0
+        else:
+            max_axis = '.yz'.index(self.direction)
+
+        # set unselected brain hemisphere activations to 0
+        if self.direction == 'l':
+            data_selection = data.copy()
```

Comment:
How should it be done? I was just trying to avoid modifying the original data "passed by reference" to the transform_to_2d function. Could I instead just do data=data to create a new function local variable?